src/apis.cpp

```diff
@@ -1180,6 +1182,58 @@ int gds_post_descriptors(size_t n_descs, gds_descriptor_t *descs, int flags)
         return ret;
 }
 
+struct mlx5_sge{
+    uint32_t byte_count;
+    uint32_t key;
+    uint64_t addr;
+};
+
+struct mlx5_send_wqe{
+    uint32_t ctrl1;
+    uint32_t qpn_ds;
+    uint64_t ctrl34;
+    uint64_t send12;
+    uint64_t send34;
+    struct mlx5_sge sge;
+};
+
+int gds_report_post(struct gds_qp *qp  /*, struct gds_send_wr* wr*/){
+    ++(qp->swq_cnt);
+    return 0;
+    /*//Smarter Alternative for cases we use larger wqes:
+    struct mlx5_send_wqe* wqe = (struct mlx5_send_wqe*)  ((char*) qp->swq + qp->swq_stride * ((qp->swq_cnt) % qp->swq_size));
+    size_t ds = (ntohl(wqe->qpn_ds) & (0x0000007f));
+    size_t wqes_per_block = (qp->swq_stride / sizeof(mlx5_sge));
+    size_t num_blocks = ds / wqes_per_block + !!(ds % wqes_per_block);
+    (qp->swq_cnt)+=num_blocks;
+    return 0;
+    */
+}
+
+int gds_query_last_info(struct gds_qp *qp, struct gds_swr_info* gds_info){
+    struct mlx5_send_wqe* wqe = (struct mlx5_send_wqe*)  ((char*) qp->swq + qp->swq_stride * ((qp->swq_cnt) % qp->swq_size));
+    gds_info->num_sge = (ntohl(wqe->qpn_ds) & (0x0000007f)) - 2;
+    struct mlx5_sge* sge = &(wqe->sge);
+    size_t blocks_per_wqe = (qp->swq_stride / sizeof(mlx5_sge));
+
+    uint16_t blocks_left = ((qp->swq_size - (qp->swq_cnt % qp->swq_size)) * qp->swq_stride) - 2;
+    //we need to monitor how many blocks we have left before wrap around.
+
+    for (size_t i = 0; i< gds_info->num_sge; ++i){
+        gds_info->sge_list[i].ptr_to_size = (uintptr_t) &(sge->byte_count);
+        gds_info->sge_list[i].ptr_to_lkey = (uintptr_t) &(sge->key);
+        gds_info->sge_list[i].ptr_to_addr = (uintptr_t) &(sge->addr);
+        gds_info->sge_list[i].offset = 0; //why is that here?
```

Comment:
what does the offset field stand for?

Reply:
@e-ago do you remember why you had offset in the first place?

Reply:
Where did you find this? Considering file https://github.com/gpudirect/libmlx5/blob/expose_send_params/src/qp.c the offset is modified here
`qp->swr_info[qp->cur_swr].sge[qp->swr_info[qp->cur_swr].cur_sge].offset = offset;`
and here
` swr_info->sge_list[j].offset = qp->swr_info[i].sge[j].offset`

Reply:
Is the offset field affecting the data written into the wqe in some way?